pkg/resource/deploy/step_generator.go

```diff
@@ -555,7 +555,79 @@ func (sg *stepGenerator) GenerateDeletes() []Step {
 			}
 		}
 	}
-	return dels
+
+	if len(targets) > 0 {
+		logging.V(7).Infof("Planner was asked to only delete '%v'", targets)
+
+		resourcesToDelete := make(map[resource.URN]bool)
+
+		for _, target := range targets {
+			current, has := sg.plan.olds[target]
+			if !has {
+				logging.V(7).Infof("Resource to delete (%v) could not be found in the stack.", target)
+				if strings.Contains(string(target), "$") {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundError(), target)
+				} else {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundDidYouForgetError(), target)
+				}
+				return nil, result.Bail()
+			}
+
+			resourcesToDelete[target] = true
+
+			// the item the user is asking to destroy may cause downstream replacements.  Clean those up
+			// as well. Use the standard delete-before-replace computation to determine the minimal
+			// set of downstream resources that are affected.
+			deps, res := sg.calculateDependentReplacements(current)
+			if res != nil {
+				return nil, res
+			}
+
+			for _, dep := range deps {
+				logging.V(7).Infof("GenerateDeletes(...): Adding dependent: %v", dep.res.URN)
+				resourcesToDelete[dep.res.URN] = true
+			}
+		}
+
+		// Also see if any resources have a resource we're deleting as a parent. If so, we'll block
+		// the delete.  It's a little painful.  But can be worked around by explicitly deleting
+		// children before parents.  Note: in almost all cases, people will want to delete children,
+		// so this restriction should not be too onerous.
+		for _, sibling := range sg.plan.prev.Resources {
```

Comment:
`sibling` is confusing here. `sibling` of what? I think `res` or similar would be clearer.

Reply:
yeah, this was a holdover from how the code was originally structured.  fixing now.